src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -224,34 +302,40 @@ public function process_directives( string $html ): string {
 			return $html;
 		}
 
-		$context_stack   = array();
-		$namespace_stack = array();
```

Comment:
One thing that I wonder about, at this top level these values always started empty at this level. I wonder if the properties should also be cleaned here. Are they ever reset to an empty array? Could we get into a state where we expect these to be empty (to start processing) but a previous call left them "dirty" with data from the last process call?

I am having some second thoughts about moving these to a class property because they are really scoped to this function. One alternative might be to pass them by reference. I think the result would be similar but they'd be clearly scoped to this function instead of class properties that only make sense at a particular time in the class.

Reply:
> I am having some second thoughts about moving these to a class property because they are really scoped to this function. One alternative might be to pass them by reference. I think the result would be similar but they'd be clearly scoped to this function instead of class properties that only make sense at a particular time in the class.

That's how it worked previously, right?

To be honest, I completely agree it would be a better approach. The main issue is how to make that temporary state available through a class method, i.e., `wp_interactivity()->get_context()`. That's why I decided to use class properties.

Would it make sense to preserve all the previous logic but store some internal references to the locally instantiated context & namespace stacks so we can use them inside `get_context()`? 🤔

Reply:
> The main issue is how to make that temporary state available through a class method, i.e., `wp_interactivity()->get_context()`

Right 😅 I forgot about this essential bit. Let's keep the class properties.

What if these stacks were created when this function runs and nulled when the function returns so they're unusable outside of directive processing?

Reply:
@DAreRodz What do you think about this part of the question?

> One thing that I wonder about, at this top level these values always started empty at this level. I wonder if the properties should also be cleaned here. Are they ever reset to an empty array? Could we get into a state where we expect these to be empty (to start processing) but a previous call left them "dirty" with data from the last process call?

Should these properties should be nullable? They'd be set to arrays during directive processing and set back to null when processing is complete. That would give us a nice way to show `_doing_it_wrong` messages when using functions the wrong way or outside of processing, it would be a `null` check.